src/kiota/Handlers/BaseKiotaCommandHandler.cs

```diff
@@ -109,7 +109,7 @@ protected async Task<KiotaSearcher> GetKiotaSearcherAsync(ILoggerFactory loggerF
         var isPatSignedIn = await patSignInCallBack(cancellationToken).ConfigureAwait(false);
         var (provider, callback) = (isDeviceCodeSignedIn, isPatSignedIn) switch
         {
-            (true, _) => (GetGitHubAuthenticationProvider(logger), deviceCodeSignInCallback),
+            (true, _) => ((IAuthenticationProvider?)GetGitHubAuthenticationProvider(logger), deviceCodeSignInCallback),
```

Comment:
when with type assertion instead?

Reply:
What do you mean by this?

Reply:
The reason I used the cast was to resolve the branch types in the switch statement. Type inference resolves the type of provider to `IAuthenticationProvider` but the last arm returns null as the provider. Without the cast, the code doesn't compile.